app/tasks/messaging.py

```diff
@@ -0,0 +1,15 @@
+from ..application import db
+from ..models.pydantic.user import User
+from ..models.orm.user import User as ORMUser
+
+
+async def send_message(ctx: dict, user_id: int, message: str):
+    # TODO: Add a messaging service.
+    orm_user: ORMUser = await ORMUser.get(user_id)
```

Comment:
I'm confused, are you reusing a database connection/pool here? If not you'll have a big overhead creating a new database connection.

Reply:
much better to store the connection pool in `ctx` to avoid ambiguous globals.

Reply:
So - startup() instantiates GINO's connection pool contained in the `db` object, as per: http://gino.fantix.pro/en/latest/tutorial.html#get-connected and http://gino.fantix.pro/en/latest/engine.html#creating-engines

The ORM models subclass that same `db` and the connection is available implicitly. Only reason I'm not using CTX here is to keep coroutines universal regardless of whether in use by arq or FastAPI, but have been giving that more thought.

Reply:
It's kind of debatable, but personally I don't like stateful global objects like connections. This is the paradigm used by aiohttp and I find it very helpful.

It makes it much easier to:
* write tests
* split code between the frontend and workers while knowing what connections you have available
* use multithreading and multiprocessing safely.
* separate your app in future if required.

global objects holding connections etc. are very simple in documentation and toy examples but lead to big headaches down the road IMHO.